contracts/utils/Giver.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.6;
+
+import "@yield-protocol/vault-interfaces/src/ICauldron.sol";
+import "@yield-protocol/vault-interfaces/src/DataTypes.sol";
+import "@yield-protocol/utils-v2/contracts/access/AccessControl.sol";
+
+/// @title A contract that allows owner of a vault to give the vault
+contract Giver is AccessControl {
+    mapping(bytes6 => bool) public ilkBlacklist;
+    ICauldron public immutable cauldron;
+
+    /// @notice Event emitted after an ilk is blacklisted
+    /// @param ilkId Ilkid to be blacklisted
+    event IlkBlacklisted(bytes6 ilkId);
+
+    constructor(ICauldron cauldron_) {
+        cauldron = cauldron_;
+    }
+
+    /// @notice Function to blacklist
+    /// @param ilkId the ilkId to be blacklisted
+    function blacklistIlk(bytes6 ilkId) external auth {
```

Comment:
Please make this a `function banIlk(bytes6 ilkId, bool set) external auth`, and `set` is used to ban/unban ilks.

Let's use `ban` instead of `blacklist`, short and clear.